src/cmds/scripts/win_postinstall.py

```diff
@@ -1,633 +1,406 @@
-        theproc = subprocess.Popen(
-            ["net", "start", "pbs_comm"], stdout=fd, stderr=fd)
-        theproc.communicate()
-        if theproc.returncode > 0:
-            print "Failed to start comm "
-            sys.exit(20)
+        print 'Successfully started %s service' % svc_name
+
+
+def register_and_start_services(username, password):
+    __svc_helper(os.path.join(pbs_sbin, 'pbs_sched.exe'), username, password)
+    __svc_helper(os.path.join(pbs_sbin, 'pbs_comm.exe'), username, password)
+    __svc_helper(os.path.join(pbs_sbin, 'pbs_mom.exe'), username, password)
+    __svc_helper(os.path.join(pbs_sbin, 'pbs_server.exe'), username, password)
+
+
+def usage():
+    prog = os.path.basename(sys.argv[0])
+    _msg = ['%s [Options]\n' % prog]
+    _msg += ['Options:\n']
+    _msg += ['    -u|--user=<username>\t- ']
+    _msg += ['specify PBS Service/database username\n']
+    _msg += ['    -p|--passwd=<password>\t- ']
+    _msg += ['specify PBS Service/database user\'s password\n']
+    _msg += ['    -h|--help\t\t\t- print help message\n']
+    print "".join(_msg)
 
 
-def main(argv):
+def main():
+    global pbs_conf_path
+    global pbs_exec
+    global pbs_home
+    global pbs_bin
+    global pbs_sbin
     username = ''
     password = ''
+    if len(sys.argv) < 2:
+        usage()
+        sys.exit(3)
     try:
-        opts, args = getopt.getopt(
-            argv, "h:u:p:", [
-                "type=", "user=", "passwd="])
-    except getopt.GetoptError:
-        print __file__ + ' -u <username> -p <password>'
-        sys.exit(2)
+        largs = ['help', 'user=', 'passwd=']
+        opts, _ = getopt.getopt(sys.argv[1:], 'hu:p:', largs)
+    except getopt.GetoptError as e:
+        print e
+        usage()
+        sys.exit(3)
     for opt, arg in opts:
-        if opt == '-h':
-            print __file__ + ' -u <username> -p <password>'
-            sys.exit()
-        elif opt in ("-u", "--user"):
+        if opt in ('-h', '--help'):
+            usage()
+            sys.exit(0)
+        elif opt in ('-u', '--user'):
             username = arg
-        elif opt in ("-p", "--passwd"):
+        elif opt in ('-p', '--passwd'):
             password = arg
-    ValidateCredentials(username, password)
-    CreatePbsConfFile()
-    InstallingVcredist()
-    InstallDataService(username, password)
-    RunPbsmkdirs()
-    RegistingAndStartingServices(username, password)
-    print "Successfully installed"
-
-
-if __name__ == "__main__":
-    install_path = os.path.dirname(os.path.realpath(__file__))
-    exec_etc_path = os.path.join("exec", "etc")
-    position_of_exec = install_path.find(exec_etc_path)
-    if position_of_exec:
-        pbs_path = install_path[:position_of_exec]
-    else:
-        print "Error while finding the PBS Pro installation path"
+        else:
+            print 'Unreocgnized option %s' % o
+            usage()
+            sys.exit(3)
+    if username is None or password is None:
```

Comment:
Both username and password are declared to '' but the check here is for None.

Reply:
@sid2364 Thanks for catching. I have addressed you comment please have a look into latest commit.

Reply:
At some point we should make the postinstall program the same one for all platforms. Ie. for Linux too. Could you file a ticket for the future?

Reply:
@subhasisb Good idea! Created [PP-1182](https://pbspro.atlassian.net/browse/PP-1182)

Reply:
Thanks for filing the issue